### PR TITLE
if_power_state_t is an enum

### PR DESCRIPTION
**Pull request by larseggert**
Fixes this error:
```
../hal/network/api/ifapi.h:261:36: error: unknown type name 'if_power_state_t'
  261 | int if_get_power_state(if_t iface, if_power_state_t* state);
      |                                    ^~~~~~~~~~~~~~~~
```

### CODE REVIEW

**Comment by avtolstoy** [comment]
I assume this issue comes up when using ifapi from C code. The correct way to resolve this is by making all these enums into typedefs:

```c++
typedef enum if_power_state_t {
    //
} if_power_state_t;
```

**Comment by larseggert** [comment]
Yes, this is from C.

**Comment by larseggert** [comment]
@avtolstoy I updated the PR

File: hal/network/api/ifapi.h [hunk 1]
```diff
@@ -29,7 +29,7 @@ extern "C" {
 typedef void* if_t;
 #endif /* IF_T_DEFINED */
 
-enum if_flags_t {
+typedef enum {
```
Comment (avtolstoy) [inline hal/network/api/ifapi.h:32]: It should be preferrable to use named enums:
```c++
typedef enum if_flags_t {
    //
} if_flags_t;
```

**Pull request closed without merging**